ubo_app/services/080-docker/docker_container.py

```diff
@@ -43,8 +55,17 @@ def find_container(client: docker.DockerClient, *, image: str) -> Container | No
 
         with contextlib.suppress(docker.errors.DockerException):
             container_image = container.image
-            if isinstance(container_image, Image) and image in container_image.tags:
-                return container
+            if isinstance(container_image, Image):
+                # Match with or without registry prefix
+                # Handles: docker.io/image:tag ↔ image:tag, ghcr.io/image ↔ image, etc.
+                matches = any(
+                    tag in image or image in tag
```

Comment:
Do you remember why do we need this?

It's too permissive and has these side effects:
1. it's too permissive and makes a piece of code that shouldn't normally work and need improvement, just work
2. it can potentially break things by accepting "pihole-branched" for "pihole" for example because "pihole" in "pihole-branched" returns true